iib/workers/config.py

```diff
@@ -237,6 +237,9 @@ class DevelopmentConfig(Config):
         "v4.15": "opm-v1.26.4",
         "v4.16": "opm-v1.40.0",
     }
+    iib_related_image_registry_replacement: dict = {
+        "registry.stage.redhat.io": "registry.stage.redhat.io"
```

Comment:
Why do we need this mapping here? The default should be set to an empty dict. We can add values in the playbooks.

Reply:
Also, shouldn't the mapping be `"registry.redhat.io": "registry.stage.redhat.io"`?

Reply:
I added it only in the DevelopmentConfig, so that if anyone needs to test with a stage bundle, they will have the config set in dev-env.

The mapping is IF a bundle is coming from "registry.stage.redhat.io", then use the "registry.stage.redhat.io" as the registry for related image pull spec as well. It is NOT just replace  registry.redhat.io with registry.stage.redhat.io.

Reply:
> I added it only in the DevelopmentConfig, so that if anyone needs to test with a stage bundle, they will have the config set in dev-env.

I would rather prefer having a default empty dict in `Config` and then anyone running the dev-env can set it to whatever they want.

> The mapping is IF a bundle is coming from "registry.stage.redhat.io", then use the "registry.stage.redhat.io" as the registry for related image pull spec as well. It is NOT just replace registry.redhat.io with registry.stage.redhat.io.

I'm not sure this is the best way to do this. What happens if I want the related images to be checked but don't want the registries replaced? Maybe the config option could be {`user`:`registry.stage.redhat.io`}? Or something else?

Reply:
>I would rather prefer having a default empty dict in Config and then anyone running the dev-env can set it to whatever they want
okay, I will update.

>What happens if I want the related images to be checked but don't want the registries replaced?
Can you give an example? The case in jira was specific for stage pushes, so stage pushes always have bundle in registry.stage.redhat.io and for stage pushes we should only check for related image availability in registry.stage.redhat.io.
 Theoretically we can use the use config as `{"iib-stage@REDHAT.COM":"registry.stage.redhat.io}` but then instead of saying check for images in the same registry as bundle, we would explicitly say check for images in this registry for this specific user and conceptually the first option sounded better to me. It would also make a meaningful test in iib-api-test as there everything comes from iib-tester

Reply:
I added the default dict in the Config but then I also wanted to understand when we chose to add the config only in Config and when do we add it in DevelopmentConfig as well.
Turns out for some configs, we need to have the default in Config but its better to set the in DevelopmentConfig as well like for iib_organization_customizations, iib_request_logs_dir, etc, so I went with same approach here. For now the changes are added without the change in dictionary mapping.

Reply:
> Can you give an example? The case in jira was specific for stage pushes, so stage pushes always have bundle in registry.stage.redhat.io and for stage pushes we should only check for related image availability in registry.stage.redhat.io.

Given the open source nature of IIB, we cannot just be thinking of use cases specific to business logic. Keeping it like this means if I am a different user trying to have the related images check without the registry replacement, that use case wouldn't be valid. 

> Theoretically we can use the use config as {user":"registry.stage.redhat.io} but then instead of saying check for images in the same registry as bundle, we would explicitly say check for images in this registry for this specific user and conceptually the first option sounded better to me. It would also make a meaningful test in iib-api-test as there everything comes from iib-tester

I'm not saying we use {user":"registry.stage.redhat.io}. It could be something else, maybe {"user": {"registry1": "registry2"}} or something completely different. I just want the use case I mentioned above to be covered since imposing some config change for everyone just because a particular user wants it sounds wrong. 

Also, if we say "check for images in the same registry as bundle" that becomes very specific and constrained. I think there's room to make the feature more generic. Like the bundle could be "registry.foo.io" while its relatedImages could be "registry.bar.io" and "quay.io". If we go with the nested dict approach, we can allow for multiple registry replacements. 


>only in Config and when do we add it in DevelopmentConfig as well.
Turns out for some configs, we need to have the default in Config but its better to set the in DevelopmentConfig as well like for iib_organization_customizations, iib_request_logs_dir, etc, so I went with same approach here. For now the changes are added without the change in dictionary mapping.

That's a good question. I don't remember the reason behind `iib_organization_customizations` being added to Dev Config. The variables in `Config` get inherited to all other classes, so setting a variable value there would mean the user doesn't have to provide a custom value if they're happy with using the default or don't have any use for that config variable. I'm ok with setting the variable in `Development Config` but as long as it doesn't have a value specific to RH.